qiskit/quantum_info/synthesis/two_qubit_decompose.py

```diff
@@ -912,23 +913,220 @@ def __call__(self, target, basis_fidelity=None, *, _num_basis_uses=None) -> Quan
+                        )
+                    elif best_nbasis == 2:
+                        return_circuit = self._get_sx_vz_2cx_efficient_euler(
+                            decomposition, target_decomposed
+                        )
+                else:
+                    raise QiskitError(
+                        "pulse_optimizer currently only works with CNOT " "entangling gate"
+                    )
+            else:
+                raise QiskitError(
+                    '"pulse_optimize" currently only works with ZSX basis '
+                    f"({self._decomposer1q.basis} used)"
+                )
+        else:
+            decomposition_euler = [self._decomposer1q._decompose(x) for x in decomposition]
+            q = QuantumRegister(2)
+            return_circuit = QuantumCircuit(q)
+            return_circuit.global_phase = target_decomposed.global_phase
+            return_circuit.global_phase -= best_nbasis * self.basis.global_phase
+            if best_nbasis == 2:
+                return_circuit.global_phase += np.pi
+            for i in range(best_nbasis):
+                return_circuit.compose(decomposition_euler[2 * i], [q[0]], inplace=True)
+                return_circuit.compose(decomposition_euler[2 * i + 1], [q[1]], inplace=True)
+                return_circuit.append(self.gate, [q[0], q[1]])
+            return_circuit.compose(decomposition_euler[2 * best_nbasis], [q[0]], inplace=True)
+            return_circuit.compose(decomposition_euler[2 * best_nbasis + 1], [q[1]], inplace=True)
         return return_circuit
 
+    def _get_sx_vz_2cx_efficient_euler(self, decomposition, target_decomposed):
+        """
+        Decomposition of SU(4) gate for device with SX, virtual RZ, and CNOT gates assuming
+        two CNOT gates are needed.
+        """
+        best_nbasis = 2  # by assumption
+        from scipy.spatial.transform import Rotation
+
+        num_1q_uni = len(decomposition)
+        dq0 = np.empty((int(num_1q_uni / 2), 3), dtype=float)
+        dq1 = np.empty((int(num_1q_uni / 2), 3), dtype=float)
+        global_phase = 0.0
+        for iqubit, idecomp in enumerate(range(0, num_1q_uni, 2)):
+            axis, angle, phase = _su2_axis_angle(decomposition[idecomp])
+            rot = Rotation.from_rotvec(axis * angle)
+            euler_angles = rot.as_euler("zxz")
+            dq0[iqubit, :] = euler_angles
+            global_phase += phase
+        for iqubit, idecomp in enumerate(range(1, num_1q_uni, 2)):
+            axis, angle, phase = _su2_axis_angle(decomposition[idecomp])
+            rot = Rotation.from_rotvec(axis * angle)
+            euler_angles = rot.as_euler("xzx")
+            dq1[iqubit, :] = euler_angles
+            global_phase += phase
+        qc = QuantumCircuit(2)
+        qc.global_phase = target_decomposed.global_phase
+        qc.global_phase -= best_nbasis * self.basis.global_phase
+
+        # TODO: make this more effecient
+        qc1q = QuantumCircuit(1)
```

Comment:
I'd appreciate a long-form comment here explaining what the strategy is. It might also help to pick more expressive variable names. `qc1q` is unpleasant, but I can guess at what it stands for as an acronym; `dq0` I have no idea.

Reply:
`dq0` was meant to denote "list of decomposition unitaries on qubit 0". I'll see if I can make a compromise in length.